pkg/operators/machine-api-operator.go

```diff
@@ -203,3 +203,58 @@ var _ = Describe("[Feature:Operators] Machine API cluster operator status should
 		Expect(framework.IsStatusAvailable(client, "machine-api")).To(BeTrue())
 	})
 })
+
+var _ = Describe("[Serial][Feature:Operators] When cluster-wide proxy is configured, Machine API cluster operator should ", func() {
+	It("create machines when configured behind a proxy", func() {
+		By("creating host for proxy")
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		proxyMachineSet, err := framework.CreateProxyMachineSet(client)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("deploying an HTTP proxy")
+		deployment, err := framework.DeployClusterProxy(client)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(framework.IsDeploymentAvailable(client, deployment.Name, deployment.Namespace)).To(BeTrue())
+
+		By("configuring cluster-wide proxy")
+		pods, err := framework.GetPods(client, map[string]string{"app": "squid"})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pods.Items) > 0).To(BeTrue())
+		proxy, err := framework.GetClusterProxy(client)
+		Expect(err).NotTo(HaveOccurred())
+		proxy.Spec.HTTPProxy = "http://" + pods.Items[0].Status.HostIP + ":80"
+		proxy.Spec.HTTPSProxy = "http://" + pods.Items[0].Status.HostIP + ":80"
```

Comment:
should this be `443` for the https proxy?

Reply:
It is non-intuitive but both HTTP and HTTPS proxy requests are proxied through HTTP.  Most HTTP proxies don't accept HTTPS connections directly to the proxy.  Additionally, the proxy I deployed for this is configured to meet this spec.

Reply:
ack, thanks!